File: java_pro/src/main/java/jv/com/tencent/cain/design_pattern/state/EveningState.java
package jv.com.tencent.cain.design_pattern.state;

/**
 * @author cainjiang
 * @date 2020/3/27
 */
public class EveningState extends State {
    @Override
    public void doWork(Work work) {
        if (work.isFinished()) {
            work.setState(new ResetState());
            work.doWork();
        } else if (work.getHour() < 21) {
            System.out.println("当前时间：" + work.getHour() + ", 属于晚上加班时间");
        } else {
            System.out.println("当前时间：" + work.getHour() + ",现在已经很晚了，今天先下班休息吧，身体最重要");
            work.setState(new ResetState());
            work.doWork();
        }
    }
}

File: java_pro/src/main/kotlin/kt/com/tencent/cain/desigin_pattern/state/EveningState.kt
package kt.com.tencent.cain.desigin_pattern.state

/**
 * @author : jiangyu
 * @date   : 2020/3/28
 * @desc   : xxx
 */
class EveningState : State() {
    override fun doWork(work: Work) {
        if (work.isFinished) {
            println("工作已经完成，可以下班了")
            work.state = ResetState()
            work.doWork()
        } else if (work.curTimeHour < 21) {
            println("当前时间：${work.curTimeHour}点，工作还未完成，需要加班")
        } else {
            println("当前时间：${work.curTimeHour}点，夜深了，注意身体，请下班休息")
            work.state = ResetState()
            work.doWork()
        }
    }
}